glide-core/src/client/value_conversion.rs

```diff
@@ -379,6 +380,101 @@ pub(crate) fn convert_to_expected_type(
+        > GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST WITHHASH
+        1) 1) "Catania"
+            2) "56.4413"
+            3) (integer) 3479447370796909
+            4) 1) "15.08726745843887329"
+                2) "37.50266842333162032"
+        to this:
+        > GEOSEARCH Sicily FROMLONLAT 15 37 BYBOX 400 400 km ASC WITHCOORD WITHDIST WITHHASH
+        1) 1) "Catania"
+            2) (double) 56.4413
+            3) (integer) 3479447370796909
+            4) 1) (double) 15.08726745843887329
+                2) (double) 37.50266842333162032
+         */
+        ExpectedReturnType::GeoSearchReturnType(with_option) => match value {
+            Value::Array(array) => {
+                if with_option == 0 {
+                    return Ok(Value::Array(array));
+                }
+
+                let mut converted_array = Vec::with_capacity(array.len());
+                for item in &array {
+                    if let Value::Array(inner_array) = item {
+                        if let Some((name, rest)) = inner_array.split_first() {
+                            if rest.len() != with_option as usize {
+                                return Err((
+                                    ErrorKind::TypeError,
+                                    "Response couldn't be converted to GeoSearchReturn type.",
+                                )
+                                    .into());
+                            }
+
+                            let rest = rest.iter().map(|v| {
+                                match v {
+                                    Value::Array(coord) => {
+                                        // This is the [lon (str), lat (str)] that should be converted into [lon (float), lat (float)].
+                                        if coord.len() != 2 {
+                                            Err((
+                                                ErrorKind::TypeError,
+                                                "Inner Array must contain exactly two elements, longitude and latitude",
+                                            ).into())
+                                        } else {
+                                            let longitude = convert_to_expected_type(
+                                                coord[0].clone(),
+                                                Some(ExpectedReturnType::Double),
+                                            )?;
+                                            let latitude = convert_to_expected_type(
+                                                coord[1].clone(),
+                                                Some(ExpectedReturnType::Double),
+                                            )?;
+                                            Ok(Value::Array(vec![longitude, latitude]))
+                                        }
+                                    }
+                                    Value::BulkString(dist) => {
```

Comment:
I think you can simplify the code and remove duplicated blocks by traversing it recursively:
```
Value::Array(array) => {
  for elem in array
    convert_to_extected_type(elem, ExpectedReturnType::GeoSearchReturnType)
  .. reassemble the array ..
}
String => convert(float)
Int => OK() // do nothing
```
This will do everything for you, but it will fail on the place name (the very first element). To avoid this in `for` loop
```
  res = convert ...
  new_elem = res.or(elem)
```
This will be colerant to FP parsing errors - but is it possible?

Reply:
even if it is, are we sure this is what we want?

this could mean an integer value could be returned as is and wont raise errors since its not an array